src/main/java/com/ruralsourcing/WeatherBot/service/HttpCalls.java

```diff
@@ -41,7 +41,12 @@ public String handleResponse(HttpResponse response) throws IOException {
 
                     if(status >= 200 && status < 300){
                         HttpEntity entity = response.getEntity();
-                        return entity != null ? EntityUtils.toString(entity) : null;
+                        if (entity != null) {
+                            String entityString = EntityUtils.toString(entity);
+                            return entityString;
+                        } else {
+                            return null;
```

Comment:
Returning `null` here isn't safe, since you pass this value into [`Gson.fromJson` in `ApiTimerTask`](https://github.com/katy-more-RSI/WeatherBot/blob/79c84ab5c0ee21a26f359bfc162d9f28a508aa56/src/main/java/com/ruralsourcing/WeatherBot/service/ApiTimerTask.java#L39). `Gson.fromJson` _is_ `null` safe, but it will return `null` and set the value in your `reference` to `null`, which is [retrieved in `SlackIntegeration`](https://github.com/katy-more-RSI/WeatherBot/blob/79c84ab5c0ee21a26f359bfc162d9f28a508aa56/src/main/java/com/ruralsourcing/WeatherBot/service/SlackIntegration.java#L39) and is later used in a [`null`-unsafe way](https://github.com/katy-more-RSI/WeatherBot/blob/79c84ab5c0ee21a26f359bfc162d9f28a508aa56/src/main/java/com/ruralsourcing/WeatherBot/service/SlackIntegration.java#L85-L93).

So all that being said, I would throw an `IOException` here instead of returning `null` (since this method already `throws IOException` that's the easiest fix). The message for the exception could be something along the lines of `No entity found in response`, or something like that.